### ISSUE TITLE
use the same openapi, json file generate ok, but yaml file generate error

### ISSUE THREAD

**Issue by liangskyli** [drwpow/openapi-typescript#1093]
**Description**

the same openapi:
when use openapi.yaml have a generate bug.
when use openapi.json no generate bug.

use openapi.yaml generate error data:
```
/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */


export interface paths {
  "/root/v4/getQueryParams1-v4": {
    get: {
      parameters: {
        query: {
          /** @description activityBases */
          activityBases: (external["components[%22schemas%22][%22ActivityBase%22]"])[];
        };
      };
      responses: {
        /** @description Success */
        200: {
          content: {
            "text/plain": string;
          };
        };
      };
    };
  };
}

export type webhooks = Record<string, never>;

export interface components {
  schemas: {
    getQueryParams1Request: {
      /** @description activityBases */
      activityBases: (external["components[%22schemas%22][%22ActivityBase%22]"])[];
    };
    ActivityBase: {
      name: string;
    } & {
      id: number;
    };
  };
  responses: never;
  parameters: never;
  requestBodies: never;
  headers: never;
  pathItems: never;
}

export type external = Record<string, never>;

export type operations = Record<string, never>;
```

| Name                 | Version                      |
|:---------------------|:-----------------------------|
| `openapi-typescript` | `6.2.4`                      |
| Node.js              | `16.14.0`                      |
| OS + version         | `macOS 13, Windows 10, etc.` |

**Reproduction**

_How can this be reproduced / when did the error occur?_

openapi.json
```
{
  "openapi": "3.1.0",
  "info": {
    "title": "custom title",
    "license": {
      "name": "This file is auto generated, do not edit!"
    },
    "version": "0.4.1"
  },
  "paths": {
    "/root/v4/getQueryParams1-v4": {
      "get": {
        "tags": ["Test4"],
        "responses": {
          "200": {
            "description": "Success",
            "content": {
              "text/plain": {
                "schema": {
                  "type": "string"
                }
              }
            }
          }
        },
        "parameters": [
          {
            "name": "activityBases",
            "in": "query",
            "required": true,
            "schema": {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/ActivityBase"
              }
            },
            "description": "activityBases"
          }
        ]
      }
    }
  },
  "components": {
    "schemas": {
      "getQueryParams1Request": {
        "type": "object",
        "properties": {
          "activityBases": {
            "description": "activityBases",
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/ActivityBase"
            }
          }
        },
        "required": ["activityBases"]
      },
      "ActivityBase": {
        "allOf": [
          {
            "type": "object",
            "properties": {
              "name": {
                "type": "string"
              }
            },
            "required": ["name"]
          },
          {
            "type": "object",
            "properties": {
              "id": {
                "type": "number"
              }
            },
            "required": ["id"]
          }
        ]
      }
    }
  },
  "tags": [
    {
      "name": "Test4",
      "description": "Test3Controller 注释\n注释3"
    }
  ]
}
```

openapi.yaml
```
openapi: 3.1.0
info:
  title: custom title
  license:
    name: This file is auto generated, do not edit!
  version: 0.4.1
paths:
  /root/v4/getQueryParams1-v4:
    get:
      tags:
        - Test4
      responses:
        '200':
          description: Success
          content:
            text/plain:
              schema:
                type: string
      parameters:
        - name: activityBases
          in: query
          required: true
          schema:
            type: array
            items: &ref_0
              $ref: '#/components/schemas/ActivityBase'
          description: activityBases
components:
  schemas:
    getQueryParams1Request:
      type: object
      properties:
        activityBases:
          description: activityBases
          type: array
          items: *ref_0
      required:
        - activityBases
    ActivityBase:
      allOf:
        - type: object
          properties:
            name:
              type: string
          required:
            - name
        - type: object
          properties:
            id:
              type: number
          required:
            - id
tags:
  - name: Test4
    description: |-
      Test3Controller 注释
      注释3
```

**Expected result**

```
/**
 * This file was auto-generated by openapi-typescript.
 * Do not make direct changes to the file.
 */


export interface paths {
  "/root/v4/getQueryParams1-v4": {
    get: {
      parameters: {
        query: {
          /** @description activityBases */
          activityBases: (components["schemas"]["ActivityBase"])[];
        };
      };
      responses: {
        /** @description Success */
        200: {
          content: {
            "text/plain": string;
          };
        };
      };
    };
  };
}

export type webhooks = Record<string, never>;

export interface components {
  schemas: {
    getQueryParams1Request: {
      /** @description activityBases */
      activityBases: (components["schemas"]["ActivityBase"])[];
    };
    ActivityBase: {
      name: string;
    } & {
      id: number;
    };
  };
  responses: never;
  parameters: never;
  requestBodies: never;
  headers: never;
  pathItems: never;
}

export type external = Record<string, never>;

export type operations = Record<string, never>;
```

**Checklist**

- [x] I’m willing to open a PR (see [CONTRIBUTING.md](https://github.com/drwpow/openapi-typescript/blob/main/CONTRIBUTING.md))

**Comment by drwpow** [comment]
I’ll take a look at this. Thank you for submitting a PR! But I fear that may not fix the underlying issue here, and I’d like to take a different approach.

**Comment by liangskyli** [comment]
The reason for this bug is:
use items: &ref_0 and items: *ref_0, Causes the ref reference in the currentSchema to point to the same address, and the currentSchema data has changed, resulting in ref transform many times！
I use JSON.parse and JSON.stringify to handle schema data before walk currentSchema to refactor code.
@drwpow : I have changed the implementation. Do you have time to take a look at this plan?

**Comment by liangskyli** [comment]
The core reason is due to changes in the value of currentSchema,
I haven't had time to deal with this in detail lately, I'm quite busy. If you have time, you can check if there's a better solution. Thank you!

**Comment by drwpow** [comment]
Thanks for the update. I plan on digging into this more this weekend myself!

**Comment by drwpow** [comment]
OK I’m digging into this for the first time, and I think I’m missing something:

```yaml
          schema:
            type: array
            items: &ref_0
              $ref: '#/components/schemas/ActivityBase'
```

How is this valid YAML?


```yaml
      properties:
        activityBases:
          description: activityBases
          type: array
          items: *ref_0
```

How is this a valid OpenAPI schema?

When I try and validate this schema, I get:

```
Error parsing /Users/drew/Sites/drwpow/openapi-typescript/packages/openapi-typescript/test.yaml: unidentified alias "ref_0" at line 36, column 24:
              items: *ref_0
```

Could you try updating your example with a YAML schema that:

- [x] is valid YAML, and
- [x] passes OpenAPI 3.1 validation? (e.g. [Swagger CLI](https://apitools.dev/swagger-cli/))?

**Comment by liangskyli** [comment]
this is valid YAML! openapi.yaml generate by js-yaml
```ts
 import { dump } from 'js-yaml'; 
dump('example data');
```
<img width="721" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/31531283/b672b73b-1b76-4586-9f16-e1c0cec1a66f">

see: https://yaml.org/spec/1.2.2/

**Comment by liangskyli** [comment]
YAML: & Node Anchors
<img width="517" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/31531283/e529c951-9a40-44a5-a864-9d706e43d00a">

**Comment by liangskyli** [comment]
use: http://nodeca.github.io/js-yaml/
<img width="1127" alt="image" src="https://github.com/drwpow/openapi-typescript/assets/31531283/cc81f71f-e276-470c-8c2b-0f6e21101424">

**Comment by liangskyli** [comment]
when i use, noRefs true, it will not use & Node Anchors generate yaml file
```ts
import { dump } from 'js-yaml';

dump({obj data}),{noRefs: true});
```

Now I have temporarily added this configuration to the generated yaml file.

**Comment by liangskyli** [comment]
Excuse me, how is the progress in handling this issue? Do you have a better plan?

**Comment by drwpow** [comment]
Hello! Yes I’ve pinpointed the issue. And I can fix it easily. Will comment thoughts on the PR.

**Comment by drwpow** [comment]
This has been fixed in `v6.2.6`!
